libs/qlik/src/visual/vl-qlik-visual.component.ts

```diff
@@ -0,0 +1,80 @@
+import { css, CSSResult, html, PropertyDeclarations, unsafeCSS } from 'lit';
+import { BaseLitElement, webComponent } from '@domg-wc/common-utilities';
+import { stardust } from '@nebula.js/stardust/types';
+import { vlElementsStyle } from '@domg-wc/elements';
+
+@webComponent('vl-qlik-visual')
+export class VlQlikVisualComponent extends BaseLitElement {
+    properties: EngineAPI.IGenericObjectProperties;
+    stardust?: stardust.Embed;
+    private type = '';
+    private qlikId = '';
+    private height = '';
+    private width? = '';
+    private additionalStyle = '';
+    private visual: (stardust.Viz | stardust.Sheet) & {
+        __DO_NOT_USE__?: any;
+        model: any;
+    };
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [vlElementsStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            type: { type: String, attribute: 'type', reflect: true },
+            qlikId: { type: String, attribute: 'qlik-id', reflect: true },
+            height: { type: String, attribute: 'height', reflect: true },
+            width: { type: String, attribute: 'width', reflect: true },
+            additionalStyle: { type: String, attribute: 'additional-style', reflect: true },
+            properties: { type: Object },
+            stardust: { type: Object },
+        };
+    }
+
+    async updated(_changedProperties) {
+        super.updated(_changedProperties)
+        if (this.visual) {
+            this.visual.destroy();
+        }
+        this.visual = await this.stardust?.render({
+            element: this.shadowRoot.getElementById(`visual-${this.qlikId}`),
+            id: this.qlikId,
+            type: this.type,
+            properties: this.properties,
+        });
+
+        this.visual?.model.on(
+            'changed',
+            (() => {
+                this.dispatchEvent(new CustomEvent('visual-changed'));
+            }).bind(this)
+        );
+
+        if (this.type === 'distributionplot') {
```

Comment:
Je gaat hier binnen `updated()` opnieuw een property wijzigen (`this.type`) dit gaat opnieuw een `updated()` triggeren - we proberen doorgaans te vermijden om properties waar Lit naar luistert tijdens lifecycle te updaten binnen de `updated()` methode.

Code werkt maar kan dit anders?

Reply:
Dit is een vergelijking, geen toewijzing

Reply:
klopt mijn excuses! :)